README.md

```diff
@@ -0,0 +1,265 @@
+
+```sh
+tp-worker-client status
+PID   NAME  STATUS  OUTLEN SUBMITTED STARTED  STOPPED   DURATION
+12345 ls1   Running 6      15:55:55  15:55:55 15:55:57  2s
+```
+
+### Client Streams Output
+
+Requires either a name or pid to be passed, like `stop` commmand does.
+
+```
+tp-worker-client output ls1
+...
+< Streamed Output >
+...
+```
+
+## Further Development / Product Ideas
+
+These are being outlined to provide potential future directions for the product,
+a place to consider where it might be appropriate to provide Interfaces rather
+than full implementations in the libraries, and anti-priorities (work that
+should be avoided at all costs, for now) for completing the task as outlined
+above.
+
+### Scanning and Obfuscating Output
+
+Scanning and Obfuscating potentially sensitive output by default would be an
+enormous win. Development of a dictionary of words to trigger obfuscation of
+following words based on word boundaries would take time and never truly be
+complete though. Also, regex's and even simple text-matching algorithms are a
+bit costly, so careful profiling and benchmarking would be necessary to select
+the best algorithms to prevent all of the server cycles from being taken up by
+scanning and obfuscation.
+
+### Privileged Elevation and Delegation Management (PEDM)
+
+To a bare minimum extent PAM and PASM are part of the above implementation, but
+there's an opportunity to do more.
+
+Include a way for a security team, manager, or other users to temporarily grant
+scoped, elevated privileges to a specific user for a specified duration. This
+could permit a user to perform sensitive, infrequent operations with the
+acknowledgement, supervision, and consent of others in the organization.
+
+### Database Storage
+
+Database job lookup pointing to output stored in block/object storage would
+help with things like incident root causing and security audits to understand
+who did what, when, and what the outcome was. It could potentially also provide
+a way for users to issue a large of volume of "fire and forget" commands that
+they intend to later retreive output for rather than babysit.
+
+Offloading streaming output to block/object storage would also reduce load on
+the server when a user is pulling down a high volume of output.
+
+Obfuscation of sensitive data would need to happen before the output is stored.
+
+### Attractive TUI Experience & JSON Web Tokens
+
+TUI (Terminal User Interface) using `bubbletea` library.
+
+Create an interactive experience in the terminal that users could keep running
+in a terminal window and use to send the commands developed for the application
+and receive/retreive the relevant outputs. Think like `k9s`.
+
+This would be especially powerful when used with a Bearer token, like a JSON
+Web Token (JWT) so that the client could send the user's entitlements to the
+server with each request via the JWT. A users's first server access in a session
```

Comment:
I know this section is only elaborating on future ideas, so I don't want to focus too much on this point, but I don't see why a JWT is necessary.

> the client could send the user's entitlements to the server with each request via the JWT

Isn't a JWT just a signed hunk of metadata that the server can first verify and then act on?

Isn't our client already providing a signed hunk of metadata to the server?